<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Omer Schwartzman" />

<meta name="date" content="2016-08-02" />

<title>umi4cPackage</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">umi4cPackage</h1>
<h4 class="author"><em>Omer Schwartzman</em></h4>
<h4 class="date"><em>2016-08-02</em></h4>



<p>UMI-4C Package is a processing and analysis pipeline for UMI-4C experiments.</p>
<div id="background" class="section level2">
<h2>Background</h2>
<p>This package is intended for processing and analysis of UMI-4C experiments. The UMI-4C protocol exploits sonication of 3C template material to facilitate molecule counting and quantitative analysis of chromatin contacts of one or several genomic loci at high resolution.</p>
<p>The two parts of the package are:</p>
<ul>
<li><strong>Processing of of raw reads (FASTQ format) to genomic tracks:</strong> <strong>Paired-end</strong> FASTQ files are parsed, splitted according to the restriction enzyme sequence (Usually DpnII) and summarized in adjacency table (<em>adj</em> file). This part uses as a bunch of <em>Perl</em> scripts wrapped in R.</li>
<li><strong>Analyzing UMI-4C genomic tracks and producing nearcis figures:</strong> Genomic tracks are dynamically processed to produce nearcis profiles at the desired genomic region specified by the user.</li>
</ul>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<div id="requirements" class="section level3">
<h3>Requirements:</h3>
<p>The pipeline is designed to run on a standard linux machine. The basic requirements are:</p>
<ul>
<li><em>Perl</em></li>
<li><em>Bowtie2</em>: <a href="http://bowtie-bio.sourceforge.net/bowtie2" class="uri">http://bowtie-bio.sourceforge.net/bowtie2</a>.</li>
<li>Reference index for <em>Bowtie2</em>.</li>
<li>R packages:
<ul>
<li><em>devtools</em>.</li>
<li><em>misha</em>.</li>
<li><em>zoo</em>.</li>
</ul></li>
</ul>
<div id="installing-misha-package" class="section level4">
<h4>Installing misha package:</h4>
<p>Since <em>misha</em> package is not in CRAN, we will need to install it directly as follow:<br />
Download the package from: <a href="http://compgenomics.weizmann.ac.il/tanay/?page_id=617" class="uri">http://compgenomics.weizmann.ac.il/tanay/?page_id=617</a>.<br />
Install the package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;misha_3.2.7.tar.gz&quot;</span>, <span class="dt">repos=</span><span class="ot">NULL</span>) <span class="co">#Installs misha package from file</span></code></pre></div>
</div>
<div id="importing-umi-4c-package" class="section level4">
<h4>Importing UMI-4C package:</h4>
<p>Download and install <em>umi4cPackage</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools::<span class="kw">install_bitbucket</span>(<span class="st">&quot;tanaylab/umi4cpackage&quot;</span>)
<span class="kw">library</span>(umi4cPackage)</code></pre></div>
<div id="preparing-genomic-database" class="section level5">
<h5>Preparing genomic database</h5>
<p>The pipeline requires a special database called <em>trackdb</em> which contains binary genomic tracks and other data, as well as restriction site database in text format called <em>redb</em>.</p>
<p>Downloading the data (~2.5Gb) is done from inside the package.<br />
We will first create a new project directory:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> my_u4c_proj</code></pre></div>
<p>From <em>R</em> we will download the data to the provided folder:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;umi4cPackage&quot;</span>)
<span class="co"># Use build 'hg19' for DpnII enzyme</span>
<span class="kw">p4cBuildRequirements</span>(<span class="dt">output_dir =</span> <span class="st">&quot;my_u4c_proj&quot;</span>, <span class="dt">genome =</span> <span class="st">&quot;hg19&quot;</span>, <span class="dt">reseq =</span> <span class="st">&quot;GATC&quot;</span>)</code></pre></div>
<p>Two directory will be generated after downloading the extracting the data:</p>
<ul>
<li><strong>my_u4c_proj/hg19/trackdb</strong><br />
</li>
<li><strong>my_u4c_proj/hg19/redb</strong></li>
</ul>
<hr />
</div>
</div>
</div>
<div id="configuration-files" class="section level3">
<h3>Configuration files</h3>
<p>The next step is to set up the <em>conf</em> files. To do so we will first need to dump templates of conf files to a conf directory:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">p4cDumpConfFiles</span>(<span class="dt">conf_dir =</span> <span class="st">&quot;my_u4c_proj/conf&quot;</span>)</code></pre></div>
<p>Now we will need to go to the supplied directory (<code>my_u4c_proj/conf</code> in our case) and complete the <em>conf</em> files.<br />
<strong>Make sure that all newlines are linux compatible (\n) and not windows (\r\n)!</strong></p>
<ul>
<li><strong>paths.conf</strong>: Set the paths to the directories and dependencies.<br />
</li>
<li><strong>samples.txt</strong>: Samples table, tab separated. One row for each experiment. Mandatory columns are:
<ul>
<li><strong>Sample_ID</strong>: Unique ID.</li>
<li><strong>Sample_name</strong>: Unique sample name. e.g., Cell type, condition etc…</li>
<li><strong>Experiment_name</strong>: A shared name for the experiment batch.<br />
</li>
<li><strong>fastqs_regex</strong>: Regular expression that captures all FASTQ files names associated with the sample. <strong>Important: the files should have the suffix *.fastq . The FASTQ filenames must also contain ‘R1’ and ‘R2’ which associate the file with the paired-end sequencing direction</strong>.</li>
<li><strong>fastqs_dir</strong>: Path to FASTQ directory.</li>
<li><strong>Bait_IDs</strong>: Unique IDs of baits associated with the experiment. A comma separated list of bait IDs from baits.txt.</li>
</ul></li>
<li><strong>baits.txt</strong>: Baits table, tab separated. One row for each bait. Mandatory columns are:
<ul>
<li><strong>Bait_ID</strong>: Unique bait ID - associates sample to bait.</li>
<li><strong>Bait_name</strong>: Unique bait name.</li>
<li><strong>Bait_seq</strong>: Sequence of the (DS) bait primer without the padding sequence. It accepts also regular expressions (e.g., AGGCA.+GGG).</li>
<li><strong>Bait_pad</strong>: Padding sequence - the sequence between the bait primer end to the bait fragment restriction site (including the restriction sequence, e.g., for DpnII: XXXXGATC).</li>
<li><strong>Bait_chr</strong>: Bait chromosome.</li>
<li><strong>Bait_coord</strong>: Bait coord. Ideally the coordinate of the nearest restriction site to the bait.</li>
</ul></li>
<li><strong>pipeline.conf</strong>: Pipeline settings. The defaults should work fine for <strong>DpnII</strong>.<br />
</li>
<li><strong>p4c.conf</strong>: Analysis settings. The defaults should work fine.</li>
</ul>
<hr />
</div>
</div>
<div id="importing-and-analyzing-umi-4c-files." class="section level2">
<h2>Importing and analyzing UMI-4C files.</h2>
<div id="initialization" class="section level3">
<h3>Initialization:</h3>
<p>We need to load the configuration files to our envirnoment by:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">p4cLoadConfFiles</span>(<span class="dt">conf_dir =</span> <span class="st">&quot;my_u4c_proj/conf&quot;</span>)</code></pre></div>
<div id="working-example" class="section level4">
<h4>Working example:</h4>
<p>Download example fastqs from: <a href="http://compgenomics.weizmann.ac.il/tanay/?page_id=617" class="uri">http://compgenomics.weizmann.ac.il/tanay/?page_id=617</a>. These files include three UMI-4C experiments done on three cell lines on the <em>ANK1</em> promoter.<br />
Extract the files to a directory.<br />
In <strong>samples.txt</strong> - change <em>fastqs_dir</em> to the working directory with the downloaded example files. Notice that the pipeline detect the files associated files by the regular expression pattern defined in <em>fastqs_regex</em>. See that <em>Bait_IDs</em> of the samples is 1.<br />
In <strong>baits.txt</strong> - Inspect that indeed the <em>Bait_ID</em> 1 is correctly filled.</p>
<div id="importing-genomic-tracks" class="section level5">
<h5>Importing genomic tracks</h5>
<p>Now we will run the pipeline on a single sample:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">p4cCreate4CseqTrack</span>(<span class="dt">sample_ids =</span> <span class="dv">1</span>)</code></pre></div>
<p>All intermediate files will be saved in the workdir that was defined in <em>TG3C.workdir</em> parameter.</p>
<p>The pipeline uses the samples and baits tables to retrieve the relevant information on experiment ID#1, and transform the raw fastq files to UMI counts (see more details on intermediate steps below). This might take some time, depending on the size of the fastq file and the machine being used. Then UMI counts are imported and saved in a special data structure we refer to as “genomic tracks”. For each sample-bait combination a genomic track will generated. In our example, a track named <code>umi4C_example_CMK_ANK1_TSS</code> will be generated. This track contains data for sample CMK and bait <em>ANK1_TSS</em>.<br />
After successfully importing the data, the pipeline will report some essential statistics.<br />
These statistics can also be accessed from 4CQC.txt in the sample workdir directory.</p>
<p>Once imported the genomic tracks are saved in our trackdb, and can be listed by:<br />
<code>r, eval=FALSE} gtrack.ls()</code></p>
</div>
</div>
<div id="analyzing-and-plotting" class="section level4">
<h4>Analyzing and plotting</h4>
<p>Now we can analyze the data:<br />
Suppose we want to create an UMI-4C profile 200kb upstream and downstream the bait in our exmaple CMK experiment. The first step is to create <code>p4cProfile object</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CMK_fc &lt;-<span class="st"> </span><span class="kw">p4cNewProfile</span>(<span class="st">&quot;umi4C_example_CMK_ANK1_TSS&quot;</span>, <span class="dt">scope_5=</span><span class="dv">200000</span>, <span class="dt">scope_3=</span><span class="dv">200000</span>)</code></pre></div>
<p>Plot nearcis profile:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(CMK_fc)</code></pre></div>
<p>This command generates a plot of the 4C profile with a smoothed trend and a domainogram. It accepts some more parameters which are described in the function documentation: <code>?plot.p4cProfile</code>.<br />
For example, if we want to save the domainogram to a png file we can do:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(CMK_fc, <span class="dt">png_fn=</span><span class="st">&quot;figs/CMK_ANK1_TSS.png&quot;</span>)</code></pre></div>
</div>
<div id="compare-the-profile-to-another-profile" class="section level4">
<h4>Compare the profile to another profile</h4>
<p>To compare two 4C experiments, we first import two other profiles (notice, this time we will do it in batch):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">p4cCreate4CseqTrack</span>(<span class="dt">sample_ids =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</code></pre></div>
<p>Two new tracks were created. To list the tracks again by <code>gtrack.ls()</code>.</p>
<p>After the tracks were imported we can produce comparative plots. For example, we will compare CMK which express <em>ANK1</em> gene to <em>DND41</em> which does not.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Generate a second p4cProfile on the same scope</span>
DND41_fc &lt;-<span class="st"> </span><span class="kw">p4cNewProfile</span>(<span class="st">&quot;umi4C_example_DND41_ANK1_TSS&quot;</span>, <span class="dt">scope_5=</span><span class="dv">200000</span>, <span class="dt">scope_3=</span><span class="dv">200000</span>)

<span class="co"># plot a comparative plot</span>
<span class="kw">plot</span>(CMK_fc, DND41_fc)</code></pre></div>
<p>Additional feature is the ability to derive mean contact intensity for defined genomic intervals. The following command will return normalized contact intensities of the two profiles, fold change, and p-value (Chi-square test)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fold_change &lt;-<span class="st"> </span><span class="kw">p4cIntervalsMean</span>(CMK_fc, DND41_fc, <span class="dt">start=</span><span class="dv">41665000</span>, <span class="dt">end=</span><span class="dv">41675000</span>)
knitr::<span class="kw">kable</span>(fold_change, <span class="dt">align =</span> <span class="st">'l'</span>, <span class="dt">digits=</span><span class="dv">2</span>)</code></pre></div>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
