<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Create a new database for umi4cPackage â€¢ umi4cPackage</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Create a new database for umi4cPackage">
<meta property="og:description" content="umi4cPackage">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">umi4cPackage</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/umi4c-usage.html">
    <span class="fas fa-road"></span>
     
    Usage
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fas fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Create a new database for umi4cPackage</h1>
                        <h4 class="author">Omer Schwartzman</h4>
                        <h4 class="author">Aviezer Lifshitz</h4>
                        <h4 class="author">Lucille Lopez-Delisle</h4>
            
            <h4 class="date">2021-06-16</h4>
      
      
      <div class="hidden name"><code>umi4c-newDB.Rmd</code></div>

    </div>

    
    
<p>UMI-4C Package is a processing and analysis pipeline for UMI-4C experiments. Some databases have been computed and are accessible via download. However, sometimes the database you need is not available and you need to create a new one. This vignette will guide you to create a new database.</p>
<div id="background" class="section level2">
<h2 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h2>
<p>To create a new genomic database, you will need:</p>
<ul>
<li>
<strong>Fasta files (one per chromosome) where file name begins with 'chr' and each chromosome name also begins with 'chr'.</strong> If you have only one file with all chromosomes the steps below will help you to convert it.</li>
<li>
<strong>Optionnally a mappability bedgraph where values goes from 0 to 5.</strong> A section is dedicated to ways to get this mappability bedgraph</li>
</ul>
</div>
<div id="get-the-inputs" class="section level2">
<h2 class="hasAnchor">
<a href="#get-the-inputs" class="anchor"></a>Get the inputs</h2>
<div id="requirements" class="section level3">
<h3 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements:</h3>
<p>The pipeline is designed to run on a standard linux machine. The basic requirements are:</p>
<ul>
<li><em>Perl</em></li>
<li>R packages:
<ul>
<li><em>devtools</em></li>
<li><em>misha</em></li>
<li><em>umi4cPackage</em></li>
</ul>
</li>
</ul>
<div id="install-the-r-dependencies" class="section level4">
<h4 class="hasAnchor">
<a href="#install-the-r-dependencies" class="anchor"></a>Install the R dependencies</h4>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Install devtools if not installed</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># Install misha if not installed</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">misha</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">'misha'</span>, repos<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/options.html">getOption</a></span><span class="op">(</span><span class="st">'repos'</span><span class="op">)</span>, <span class="st">'https://tanaylab.github.io/repo'</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># Install umi4cPackage if not installed</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">umi4cPackage</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"tanaylab/umi4cpackage"</span>, vignette <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
</div>
<div id="fasta-files" class="section level3">
<h3 class="hasAnchor">
<a href="#fasta-files" class="anchor"></a>Fasta files</h3>
<p>Most of genomes are available on UCSC where you can download them through the <a href="http://hgdownload.cse.ucsc.edu/downloads.html">download page</a> in the section <strong>Genome sequence files and select annotations</strong>. You can also download them by the link:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">genome=</span>mm10
<span class="fu">wget</span> http://hgdownload.cse.ucsc.edu/goldenPath/<span class="va">${genome}</span>/bigZips/<span class="va">${genome}</span>.fa.gz</code></pre></div>
<p>If you have a fasta file from another source it is also find.</p>
<p>If the name of the chromosome does not begin by 'chr' you will need to add it by:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">zcat</span> original.fa.gz <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'$0~/^&gt;/{gsub("&gt;","&gt;chr", $0)}{print}'</span> <span class="kw">|</span> <span class="fu">gzip</span> <span class="op">&gt;</span> <span class="va">${genome}</span>.fa.gz</code></pre></div>
<p>To create a new database, you need one file per chromosome. If you have only one file, you can split your file using awk:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">genome=</span>mm10
<span class="fu">mkdir</span> -p <span class="va">$genome</span>
<span class="fu">zcat</span> <span class="va">${genome}</span>.fa.gz <span class="kw">|</span> <span class="fu">awk</span> -v dir=<span class="va">$genome</span> <span class="st">'$0~"^&gt;"{id = $1; gsub("^&gt;", "", id)}{print &gt;&gt; dir"/"id".fa"}'</span></code></pre></div>
<p>In some assemblies there are a lot of contigs or there are alternate contigs and you may be interested in keeping only some of them (for example only the numbered chromosome, the sexual and the mitochondrion).</p>
<p>You can remove the one you don't want from the new folder created, for example:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">rm</span> <span class="va">${genome}</span>/*random
<span class="fu">rm</span> <span class="va">${genome}</span>/chrUn*
<span class="fu">rm</span> <span class="va">${genome}</span>/*_alt</code></pre></div>
</div>
<div id="build-a-mappability-bedgraph" class="section level3">
<h3 class="hasAnchor">
<a href="#build-a-mappability-bedgraph" class="anchor"></a>Build a mappability bedgraph</h3>
<p>As written above, you can provide a mappability bedgraph. This is not mandatory, alternatively you can arbitrary decide that all genome is fully mappable.</p>
<p>One way to get the mappability track is to use <a href="Link%20to%20the%20home%20page">gem2</a>. You can install it through conda:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">conda</span> install -c bioconda gem2</code></pre></div>
<p>or download the source:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">wget</span> https://paoloribeca.science/gem/gem2.INEEDTOCHECK.Linux-amd64.tar.xz
<span class="fu">tar</span> xvf gem2.INEEDTOCHECK.Linux-amd64.tar.xz
<span class="bu">export</span> <span class="va">PATH=$PATH</span>:<span class="va">$PWD</span>/gem2.INEEDTOCHECK.Linux-amd64</code></pre></div>
<p>You need to have a single fasta file. If you selected only some chromosomes, you need to concatenate them:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="fu">cat</span> <span class="va">${genome}</span>/*.fa <span class="op">&gt;</span> <span class="va">${genome}</span>_simplified.fa</code></pre></div>
<p>Then you need to create an index:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">nbThreads=</span>28
<span class="va">fastaFile=${genome}</span>_simplified.fa <span class="co"># or ${genome}.fa.gz if you did not select chromosomes</span>
<span class="ex">gem-indexer</span> -T <span class="va">${nbThreads}</span> -c dna -i <span class="va">${fastaFile}</span> -o <span class="va">${genome}</span>_index</code></pre></div>
<p>You need to decide a size for the kmer to compute mappability (we recommand 50).</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">kmer=</span>50
<span class="ex">gem-mappability</span> -T <span class="va">${nbThreads}</span> -I <span class="va">${genome}</span>_index.gem -l <span class="va">${kmer}</span> -o <span class="va">${genome}</span>_<span class="va">${kmer}</span></code></pre></div>
<p>Finally you need to convert the mappability file to bedgraph</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">gem-2-bed</span> mappability -I <span class="va">${genome}</span>_index.gem -i <span class="va">${genome}</span>_<span class="va">${kmer}</span>.mappability -o <span class="va">${genome}</span>_<span class="va">${kmer}</span></code></pre></div>
</div>
</div>
<div id="build-the-genomic-database" class="section level2">
<h2 class="hasAnchor">
<a href="#build-the-genomic-database" class="anchor"></a>Build the genomic database</h2>
<p>The pipeline requires a special database called <em>trackdb</em> which contains binary genomic tracks and other data, as well as restriction site database in text format called <em>redb</em>.</p>
<div id="initiate-the-database" class="section level3">
<h3 class="hasAnchor">
<a href="#initiate-the-database" class="anchor"></a>Initiate the database</h3>
<p>From <em>R</em> we will create a genome database (the <em>trackdb</em> folder) using individual fasta files. All fasta files must begin by "chr".</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"misha"</span><span class="op">)</span>
<span class="va">genome</span> <span class="op">&lt;-</span> <span class="st">"mm10"</span>
<span class="fu"><a href="https://rdrr.io/pkg/misha/man/gdb.create.html">gdb.create</a></span><span class="op">(</span><span class="st">"my_genome/trackdb"</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">genome</span>, <span class="st">"chr*.fa"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>You set this new database as your current database:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/misha/man/gdb.init.html">gdb.init</a></span><span class="op">(</span><span class="st">"my_genome/trackdb"</span><span class="op">)</span></code></pre></div>
</div>
<div id="put-mappability-track" class="section level3">
<h3 class="hasAnchor">
<a href="#put-mappability-track" class="anchor"></a>Put mappability track</h3>
<p>We will now create a track with the mappability. It can be very long if you have a big genome...</p>
<ul>
<li>If you did not compute any mappability bedgraph:</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/misha/man/gtrack.create.html">gtrack.create</a></span><span class="op">(</span>track<span class="op">=</span><span class="st">"my_mappab"</span>, description<span class="op">=</span><span class="st">"mappab set to 5"</span>,
              expr<span class="op">=</span><span class="st">"5"</span>, iterator <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">ALLGENOME</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">end</span><span class="op">)</span>,
              band <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<ul>
<li>If you computed the mappability with gem or any other way and you have a bedgraph:</li>
</ul>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mapab_fn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">genome</span>, <span class="st">"_50.bg"</span><span class="op">)</span>
<span class="co"># Create a track from the bedgraph</span>
<span class="fu"><a href="https://rdrr.io/pkg/misha/man/gtrack.import.html">gtrack.import</a></span><span class="op">(</span>track<span class="op">=</span><span class="st">"mappab_from_file"</span>, description<span class="op">=</span><span class="st">"mappability from gem"</span>,
              file<span class="op">=</span><span class="va">mapab_fn</span>, <span class="fl">10</span><span class="op">)</span>
<span class="co"># Get the maxiumum value (1 with gem):</span>
<span class="va">max.value</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/misha/man/gsummary.html">gsummary</a></span><span class="op">(</span><span class="st">"mappab_from_file"</span><span class="op">)</span><span class="op">[</span><span class="st">"Max"</span><span class="op">]</span>
<span class="co"># Create a new track with the values scaled to get max at 5:</span>
<span class="fu"><a href="https://rdrr.io/pkg/misha/man/gtrack.create.html">gtrack.create</a></span><span class="op">(</span>track<span class="op">=</span><span class="st">"my_mappab"</span>, description<span class="op">=</span><span class="st">"mappab from gem scaled to 5"</span>,
              expr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"mappab_from_file * 5 /"</span>, <span class="va">max.value</span><span class="op">)</span>, iterator <span class="op">=</span> <span class="fl">10</span>,
              band <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
</div>
<div id="create-the-redb-tracks" class="section level3">
<h3 class="hasAnchor">
<a href="#create-the-redb-tracks" class="anchor"></a>Create the redb tracks</h3>
<div id="build-the-chromosome_key-table" class="section level4">
<h4 class="hasAnchor">
<a href="#build-the-chromosome_key-table" class="anchor"></a>Build the chromosome_key table</h4>
<p>To create a redb you need to provide a tabulation-delimited file which provide the list of chromosomes and the path to the sequence in fasta format.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># You need to adapt here where your individual fasta are:</span>
<span class="va">fasta.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">genome</span><span class="op">)</span><span class="op">)</span>
<span class="va">seq.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">fasta.files</span>, <span class="kw">function</span><span class="op">(</span><span class="va">fn</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"^chr"</span>, <span class="st">""</span>, 
                                                 <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\.fa$"</span>, <span class="st">""</span>, <span class="va">fn</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="va">seq.key.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>names <span class="op">=</span> <span class="va">seq.names</span>,
                         files <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">genome</span>, <span class="va">fasta.files</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">seq.key.df</span>, <span class="st">"chrom_key.txt"</span>, sep <span class="op">=</span><span class="st">"\t"</span>,
            row.names <span class="op">=</span> <span class="cn">F</span>, col.names <span class="op">=</span> <span class="cn">F</span>, quote <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
</div>
<div id="configuration-file" class="section level4">
<h4 class="hasAnchor">
<a href="#configuration-file" class="anchor"></a>Configuration file</h4>
<p>The next step is to set up the <em>conf</em> file. To do so we will first need to dump templates of conf files to a conf directory:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">umi4cPackage</span><span class="op">)</span>
<span class="fu"><a href="../reference/p4cDumpConfFiles.html">p4cDumpConfFiles</a></span><span class="op">(</span>conf_dir <span class="op">=</span> <span class="st">"my_genome/conf"</span><span class="op">)</span></code></pre></div>
<p>Now we will need to go to the supplied directory (<code>my_genome/conf</code> in our case) and complete the <em>redb.conf</em> file.<br><strong>Make sure that all newlines are linux compatible (\n) and not windows (\r\n)!</strong></p>
<ul>
<li>
<strong>TG3C.mapab_track</strong>: Set the name of the mappability track</li>
<li>
<strong>TG3C.re_workdir</strong>: Set the path to the redb directory</li>
<li>
<strong>TG3C.chrom_seq_key</strong>: Set the path to the chomosome_key table created above.</li>
</ul>
</div>
<div id="create-the-db" class="section level4">
<h4 class="hasAnchor">
<a href="#create-the-db" class="anchor"></a>Create the db</h4>
<p>Now you need to create a database specific to the restriction enzyme you used.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For example, DpnII:</span>
<span class="fu"><a href="../reference/gtrack.create_redb_tracks.html">gtrack.create_redb_tracks</a></span><span class="op">(</span>re_seq<span class="op">=</span><span class="st">"GATC"</span>, redb_params_fn<span class="op">=</span><span class="st">"my_genome/conf/redb.conf"</span><span class="op">)</span></code></pre></div>
<p>After this, you should have two directories that contains all files necessary to do the umi4c analysis:</p>
<ul>
<li>
<strong>my_genome/trackdb</strong><br>
</li>
<li><strong>my_genome/redb</strong></li>
</ul>
<p>You can keep a copy of this new database, for example creating an archive:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">genome=</span>mm10
<span class="fu">tar</span> -zcvf <span class="va">${genome}</span>_umi4c_db.tar.gz my_genome/redb my_genome/trackdb</code></pre></div>
<p>When you want to use this new db, don't forget to use the good path in <strong>paths.conf</strong> file for variables <strong>TG3C.trackdb</strong> and <strong>TG3C.redb</strong>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Amos Tanay, Omer Schwartzman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
